libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -354,10 +357,16 @@ void VoteManager::cleanupVotes(uint64_t pbft_round) {
     UniqueLock lock(verified_votes_access_);
     auto it = verified_votes_.begin();
     while (it != verified_votes_.end() && it->first < pbft_round) {
-      for (auto const& step : it->second) {
-        for (auto const& voted_value : step.second) {
-          for (auto const& v : voted_value.second.second) {
+      for (const auto& step : it->second) {
+        for (const auto& voted_value : step.second) {
+          for (const auto& v : voted_value.second.second) {
             db_->removeVerifiedVoteToBatch(v.first, batch);
+            LOG(log_dg_) << "Remove verified vote " << v.first << " for round " << v.second->getRound()
+                         << ". PBFT round " << pbft_round;
+            if (v.second->getType() == cert_vote_type) {
+              // The verified vote may be reward vote
+              addRewardVote(v.second);
```

Comment:
Can you pls explain why we add reward vote here ?

Reply:
I explained already. https://github.com/Taraxa-project/taraxa-node/pull/1670#discussion_r860387304

Reply:
and I commented your explanation, but you did not answer... still dont understand why it is done here

Reply:
Here is an example why need add the verified votes to reward votes queue:

In certify stage, node receive enough cert votes and finalize PBFT block B1 with cert votes {v1, v2, v3}
Then node goes to next voting stage in the same round, and node receive additional cert vote v4 for B1. node will add v4 in votes queue here https://github.com/Taraxa-project/taraxa-node/blob/reward_cert_votes/libraries/core_libs/network/src/tarcap/packets_handlers/vote_packet_handler.cpp#L55
node receives enough next votes and goes to next PBFT round. node will clean up all votes (include v4) < current PBFT round here https://github.com/Taraxa-project/taraxa-node/blob/reward_cert_votes/libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp#L275
We cannot delete v4 here, need add v4 in reward votes